## Final Project Submission

Please fill out:
* Student name: Lucy Wanjiru Njoroge
* Student pace: self paced / part time / full time : full time
* Scheduled project review date/time: 
* Instructor name:
* Blog post URL:


In [1]:
# Your code here - remember to use markdown cells for comments as well!bb


## BUSINESS UNDERSTANDING

Microsoft , an American multinational technology company, wishes to create original video content , which has become quite popular . The company wishes to achieve this by creating a new movie studio tasked with movie production.However, they do not know anything about creating movies and need insight on the kind of genres they will need to produce based on certain factors. This project will entail the analysis of  data provided from various sources such as IMDB and rotten tomatoes  to help generate insight on this project so that the genres recommended at the end of the analysis will be the best fit in terms of Box office performance

This analysis will revolve around finding the best genre of movies that the new Microsoft Movie studio can produce. The project will centre its analysis based on movie ratings , the gross money returns as well as do an in depth analysis on the relationship between Box office performance and the production costs to get an approximated project budget.

## DATA UNDERSTANDING

In [2]:
##Importing the necessary libraries
import pandas as pd
import matplotlib as plt
%matplotlib inline

### 1. movie_budget_df

In [3]:
#acccessing the movie_budget_df dataframe 
movie_budget_df = pd.read_csv("zippedData/tn.movie_budgets.csv.gz")
movie_budget_df

id  release_date                                        movie  \
0      1  Dec 18, 2009                                       Avatar   
1      2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2      3   Jun 7, 2019                                 Dark Phoenix   
3      4   May 1, 2015                      Avengers: Age of Ultron   
4      5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   
...   ..           ...                                          ...   
5777  78  Dec 31, 2018                                       Red 11   
5778  79   Apr 2, 1999                                    Following   
5779  80  Jul 13, 2005                Return to the Land of Wonders   
5780  81  Sep 29, 2015                         A Plague So Pleasant   
5781  82   Aug 5, 2005                            My Date With Drew   

     production_budget domestic_gross worldwide_gross  
0         $425,000,000   $760,507,625  $2,776,345,279  
1         $410,600,000   $241,063,875  $1,045,663,875  
2         $350,000,000    $42,762,350    $149,762,350  
3         $330,600,000   $459,005,868  $1,403,013,963  
4         $317,000,000   $620,181,382  $1,316,721,747  
...                ...            ...             ...  
5777            $7,000             $0              $0  
5778            $6,000        $48,482        $240,495  
5779            $5,000         $1,338          $1,338  
5780            $1,400             $0              $0  
5781            $1,100       $181,041        $181,041  

[5782 rows x 6 columns]

In [10]:
movie_budget_df.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

In [11]:
movie_budget_df.tail()

id  release_date                          movie production_budget  \
5777  78  Dec 31, 2018                         Red 11            $7,000   
5778  79   Apr 2, 1999                      Following            $6,000   
5779  80  Jul 13, 2005  Return to the Land of Wonders            $5,000   
5780  81  Sep 29, 2015           A Plague So Pleasant            $1,400   
5781  82   Aug 5, 2005              My Date With Drew            $1,100   

     domestic_gross worldwide_gross  
5777             $0              $0  
5778        $48,482        $240,495  
5779         $1,338          $1,338  
5780             $0              $0  
5781       $181,041        $181,041

In [12]:
movie_budget_df.describe()

id
count  5782.000000
mean     50.372363
std      28.821076
min       1.000000
25%      25.000000
50%      50.000000
75%      75.000000
max     100.000000

In [8]:
movie_budget_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


This dataset has 5782 rows and 6 columns with no missing values

In [10]:
#This also gives the number of rows and columns respectively in a tuple
movie_budget_df.shape

(5782, 6)

In [35]:
## Relationship between the variables
movie_budget_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


The columns in the dataset are id,an integer value, release_date, movie, production_budget, domestic_gross and worldwide_gross which at this time are objects

The movie columns gives a list of the movies released , the production_budget is the budget cost of movie production and the domestic_gross and worldwide_gross columns are the income the movie has generated domestically and worldwide respectively. The id column is a way of identifying the rows

### 2. Im.db_df dataset

#### 2.1.Table1 :movie_ratings table

In [11]:
#connecting to the database
import sqlite3
conn = sqlite3.connect("zippedData/im.db/im.db")

In [12]:
#select all columns in the table
ratings_df = pd.read_sql("SELECT * FROM movie_ratings;" , conn)
ratings_df

movie_id  averagerating  numvotes
0      tt10356526            8.3        31
1      tt10384606            8.9       559
2       tt1042974            6.4        20
3       tt1043726            4.2     50352
4       tt1060240            6.5        21
...           ...            ...       ...
73851   tt9805820            8.1        25
73852   tt9844256            7.5        24
73853   tt9851050            4.7        14
73854   tt9886934            7.0         5
73855   tt9894098            6.3       128

[73856 rows x 3 columns]

In [13]:
ratings_df.head()

movie_id  averagerating  numvotes
0  tt10356526            8.3        31
1  tt10384606            8.9       559
2   tt1042974            6.4        20
3   tt1043726            4.2     50352
4   tt1060240            6.5        21

In [14]:
ratings_df.tail()

movie_id  averagerating  numvotes
73851  tt9805820            8.1        25
73852  tt9844256            7.5        24
73853  tt9851050            4.7        14
73854  tt9886934            7.0         5
73855  tt9894098            6.3       128

In [15]:
ratings_df.describe()

averagerating      numvotes
count   73856.000000  7.385600e+04
mean        6.332729  3.523662e+03
std         1.474978  3.029402e+04
min         1.000000  5.000000e+00
25%         5.500000  1.400000e+01
50%         6.500000  4.900000e+01
75%         7.400000  2.820000e+02
max        10.000000  1.841066e+06

In [16]:
ratings_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 73856 entries, 0 to 73855
Data columns (total 3 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   movie_id       73856 non-null  object 
 1   averagerating  73856 non-null  float64
 2   numvotes       73856 non-null  int64  
dtypes: float64(1), int64(1), object(1)
memory usage: 1.7+ MB


 This data frame has 73856 rows and 3 columns

In [18]:
#Also gives the number of rows and columns respectively in a tuple
ratings_df.shape

(73856, 3)

#### Relationship between the variables
The movie_id column will be used to identify the movie , while the averagerating column gives the average rating based on the number of votes in the numvotes column


#### A.Table2 :movie_basics

In [19]:
movie_basics_df = pd.read_sql("SELECT * FROM movie_basics;" , conn)
movie_basics_df

movie_id                                primary_title  \
0       tt0063540                                    Sunghursh   
1       tt0066787              One Day Before the Rainy Season   
2       tt0069049                   The Other Side of the Wind   
3       tt0069204                              Sabse Bada Sukh   
4       tt0100275                     The Wandering Soap Opera   
...           ...                                          ...   
146139  tt9916538                          Kuambil Lagi Hatiku   
146140  tt9916622  Rodolpho Teóphilo - O Legado de um Pioneiro   
146141  tt9916706                              Dankyavar Danka   
146142  tt9916730                                       6 Gunn   
146143  tt9916754               Chico Albuquerque - Revelações   

                                     original_title  start_year  \
0                                         Sunghursh        2013   
1                                   Ashad Ka Ek Din        2019   
2                        The Other Side of the Wind        2018   
3                                   Sabse Bada Sukh        2018   
4                             La Telenovela Errante        2017   
...                                             ...         ...   
146139                          Kuambil Lagi Hatiku        2019   
146140  Rodolpho Teóphilo - O Legado de um Pioneiro        2015   
146141                              Dankyavar Danka        2013   
146142                                       6 Gunn        2017   
146143               Chico Albuquerque - Revelações        2013   

        runtime_minutes                genres  
0                 175.0    Action,Crime,Drama  
1                 114.0       Biography,Drama  
2                 122.0                 Drama  
3                   NaN          Comedy,Drama  
4                  80.0  Comedy,Drama,Fantasy  
...                 ...                   ...  
146139            123.0                 Drama  
146140              NaN           Documentary  
146141              NaN                Comedy  
146142            116.0                  None  
146143              NaN           Documentary  

[146144 rows x 6 columns]

In [20]:
movie_basics_df.head()

movie_id                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   

   start_year  runtime_minutes                genres  
0        2013            175.0    Action,Crime,Drama  
1        2019            114.0       Biography,Drama  
2        2018            122.0                 Drama  
3        2018              NaN          Comedy,Drama  
4        2017             80.0  Comedy,Drama,Fantasy

In [21]:
movie_basics_df.tail()

movie_id                                primary_title  \
146139  tt9916538                          Kuambil Lagi Hatiku   
146140  tt9916622  Rodolpho Teóphilo - O Legado de um Pioneiro   
146141  tt9916706                              Dankyavar Danka   
146142  tt9916730                                       6 Gunn   
146143  tt9916754               Chico Albuquerque - Revelações   

                                     original_title  start_year  \
146139                          Kuambil Lagi Hatiku        2019   
146140  Rodolpho Teóphilo - O Legado de um Pioneiro        2015   
146141                              Dankyavar Danka        2013   
146142                                       6 Gunn        2017   
146143               Chico Albuquerque - Revelações        2013   

        runtime_minutes       genres  
146139            123.0        Drama  
146140              NaN  Documentary  
146141              NaN       Comedy  
146142            116.0         None  
146143              NaN  Documentary

In [22]:
movie_basics_df.describe()

start_year  runtime_minutes
count  146144.000000    114405.000000
mean     2014.621798        86.187247
std         2.733583       166.360590
min      2010.000000         1.000000
25%      2012.000000        70.000000
50%      2015.000000        87.000000
75%      2017.000000        99.000000
max      2115.000000     51420.000000

In [23]:
movie_basics_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 146144 entries, 0 to 146143
Data columns (total 6 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   movie_id         146144 non-null  object 
 1   primary_title    146144 non-null  object 
 2   original_title   146123 non-null  object 
 3   start_year       146144 non-null  int64  
 4   runtime_minutes  114405 non-null  float64
 5   genres           140736 non-null  object 
dtypes: float64(1), int64(1), object(4)
memory usage: 6.7+ MB


This dataset has 146144 rows and 6 columns. Original_title,runtime_minutes and genres have missing values

In [24]:
movie_basics_df.shape

(146144, 6)

#### Relationship between variables
This dataset has the columns: movie_id which helps identify the variable name in the primary_title and original_title columns, The primary_title and original_title columns give the names  of the movies, the start_year column : year that the movie either started running or was produces , runtime_minutes: The duration of the movie and the genres column :the movie type.

## DATA PREPARATION & CLEANING

 In this section , i will be checking for null and missing values  as well as duplicates and outliers.
 I will begin by merging my datasets and carrying out my data cleaning in the combined table. The data cleaning activities will involve getting rid of missing values, null values, duplicates and the outliers.

#### MERGING

### Begin by merging  Table 2 of the Im.db_df  dataset and the movie_df dataframe

In [25]:
merge_df = movie_basics_df.merge(movie_budget_df , how = "inner", left_on = "primary_title" , right_on = "movie")
merge_df

#The type of merge conducted is the inner merge based on the primary key "movie" and foreign key "primary_title " in the im.db dataset

movie_id  primary_title original_title  start_year  runtime_minutes  \
0     tt0249516     Foodfight!     Foodfight!        2012             91.0   
1     tt0293429  Mortal Kombat  Mortal Kombat        2021              NaN   
2     tt0326592  The Overnight  The Overnight        2010             88.0   
3     tt3844362  The Overnight  The Overnight        2015             79.0   
4     tt0337692    On the Road    On the Road        2012            124.0   
...         ...            ...            ...         ...              ...   
3810  tt9678962           Fuel           Fuel        2018              NaN   
3811  tt9729206          Diner          Diner        2019              NaN   
3812  tt9805168        Traitor        Traitor        2015            110.0   
3813  tt9844102            Ray            Ray        2018            111.0   
3814  tt9893078        Sublime        Sublime        2019              NaN   

                        genres  id  release_date          movie  \
0      Action,Animation,Comedy  26  Dec 31, 2012     Foodfight!   
1     Action,Adventure,Fantasy  10  Aug 18, 1995  Mortal Kombat   
2                         None  21  Jun 19, 2015  The Overnight   
3               Comedy,Mystery  21  Jun 19, 2015  The Overnight   
4      Adventure,Drama,Romance  17  Mar 22, 2013    On the Road   
...                        ...  ..           ...            ...   
3810         Documentary,Sport  35  Nov 14, 2008           Fuel   
3811                     Crime  11   Apr 2, 1982          Diner   
3812      Action,Drama,Romance  25  Aug 27, 2008        Traitor   
3813                     Crime   8  Oct 29, 2004            Ray   
3814               Documentary  30  Mar 13, 2007        Sublime   

     production_budget domestic_gross worldwide_gross  
0          $45,000,000             $0         $73,706  
1          $20,000,000    $70,433,227    $122,133,227  
2             $200,000     $1,109,808      $1,165,996  
3             $200,000     $1,109,808      $1,165,996  
4          $25,000,000       $720,828      $9,313,302  
...                ...            ...             ...  
3810        $2,500,000       $174,255        $174,255  
3811        $5,000,000    $12,592,907     $12,592,907  
3812       $22,000,000    $23,530,831     $27,882,226  
3813       $40,000,000    $75,305,995    $124,823,094  
3814        $1,800,000             $0              $0  

[3815 rows x 12 columns]

The merged dataset between Table 2 of the Im.db_ dataset has been assigned the variable name merge_df. The next step will be to merge the dataset merge_df with Table 1 of the Im.db dataset. The merge was done in this order so as not to compromise the primary and foreign key values in the movie_df dataframe and Table 2 of the Im.db_dataset and limit the number of rows for the analysis. 

###  Merging the merge_df dataset with Table 1 of the im.db_df dataset

In [26]:
final_merge_df = merge_df.merge(ratings_df , how = "inner")
final_merge_df

movie_id  primary_title   original_title  start_year  runtime_minutes  \
0     tt0249516     Foodfight!       Foodfight!        2012             91.0   
1     tt0326592  The Overnight    The Overnight        2010             88.0   
2     tt3844362  The Overnight    The Overnight        2015             79.0   
3     tt0337692    On the Road      On the Road        2012            124.0   
4     tt4339118    On the Road      On the Road        2014             89.0   
...         ...            ...              ...         ...              ...   
2870  tt8976472    Never Again  Ei koskaan enää        2018             69.0   
2871  tt8680254    Richard III      Richard III        2016              NaN   
2872  tt8824064         Heroes           Heroes        2019             88.0   
2873  tt9024106      Unplanned        Unplanned        2019            106.0   
2874  tt9248762  The Terrorist    The Terrorist        2018              NaN   

                       genres  id  release_date          movie  \
0     Action,Animation,Comedy  26  Dec 31, 2012     Foodfight!   
1                        None  21  Jun 19, 2015  The Overnight   
2              Comedy,Mystery  21  Jun 19, 2015  The Overnight   
3     Adventure,Drama,Romance  17  Mar 22, 2013    On the Road   
4                       Drama  17  Mar 22, 2013    On the Road   
...                       ...  ..           ...            ...   
2870        Documentary,Music  47  Jul 12, 2002    Never Again   
2871                    Drama  65  Dec 29, 1995    Richard III   
2872              Documentary  12  Oct 24, 2008         Heroes   
2873          Biography,Drama  33  Mar 29, 2019      Unplanned   
2874                 Thriller  48  Jan 14, 2000  The Terrorist   

     production_budget domestic_gross worldwide_gross  averagerating  numvotes  
0          $45,000,000             $0         $73,706            1.9      8248  
1             $200,000     $1,109,808      $1,165,996            7.5        24  
2             $200,000     $1,109,808      $1,165,996            6.1     14828  
3          $25,000,000       $720,828      $9,313,302            6.1     37886  
4          $25,000,000       $720,828      $9,313,302            6.0         6  
...                ...            ...             ...            ...       ...  
2870          $500,000       $307,631        $308,793            6.6        27  
2871        $9,200,000     $2,684,904      $4,199,334            9.1        28  
2872          $400,000       $655,538        $655,538            7.3         7  
2873        $6,000,000    $18,107,621     $18,107,621            6.3      5945  
2874           $25,000       $195,043        $195,043            6.0         6  

[2875 rows x 14 columns]

The final merge dataframe has been assigned the variable name final_merge_df, we can now begin with the actual data cleaning

###  Checking for null and missing values

In [28]:
#There are missing values in the runtime_minutes column as well as the genres column
final_merge_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2875 entries, 0 to 2874
Data columns (total 14 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   movie_id           2875 non-null   object 
 1   primary_title      2875 non-null   object 
 2   original_title     2875 non-null   object 
 3   start_year         2875 non-null   int64  
 4   runtime_minutes    2757 non-null   float64
 5   genres             2867 non-null   object 
 6   id                 2875 non-null   int64  
 7   release_date       2875 non-null   object 
 8   movie              2875 non-null   object 
 9   production_budget  2875 non-null   object 
 10  domestic_gross     2875 non-null   object 
 11  worldwide_gross    2875 non-null   object 
 12  averagerating      2875 non-null   float64
 13  numvotes           2875 non-null   int64  
dtypes: float64(2), int64(3), object(9)
memory usage: 336.9+ KB


In [29]:
#The number of null values in each column can also be obtained by using the .isna() and .sum() methods
final_merge_df.isna().sum()

movie_id               0
primary_title          0
original_title         0
start_year             0
runtime_minutes      118
genres                 8
id                     0
release_date           0
movie                  0
production_budget      0
domestic_gross         0
worldwide_gross        0
averagerating          0
numvotes               0
dtype: int64

Once the columns with null values have been identified, based on the number of null values, I chose to drop the release_date column  since i will not be using it in my analysis and dropped the genre rows with missing values.

In [30]:
# Dropping the run_time minutes column
final_merge_df.drop(columns = "runtime_minutes" , axis = 1 , inplace = True)

In [31]:
final_merge_df

movie_id  primary_title   original_title  start_year  \
0     tt0249516     Foodfight!       Foodfight!        2012   
1     tt0326592  The Overnight    The Overnight        2010   
2     tt3844362  The Overnight    The Overnight        2015   
3     tt0337692    On the Road      On the Road        2012   
4     tt4339118    On the Road      On the Road        2014   
...         ...            ...              ...         ...   
2870  tt8976472    Never Again  Ei koskaan enää        2018   
2871  tt8680254    Richard III      Richard III        2016   
2872  tt8824064         Heroes           Heroes        2019   
2873  tt9024106      Unplanned        Unplanned        2019   
2874  tt9248762  The Terrorist    The Terrorist        2018   

                       genres  id  release_date          movie  \
0     Action,Animation,Comedy  26  Dec 31, 2012     Foodfight!   
1                        None  21  Jun 19, 2015  The Overnight   
2              Comedy,Mystery  21  Jun 19, 2015  The Overnight   
3     Adventure,Drama,Romance  17  Mar 22, 2013    On the Road   
4                       Drama  17  Mar 22, 2013    On the Road   
...                       ...  ..           ...            ...   
2870        Documentary,Music  47  Jul 12, 2002    Never Again   
2871                    Drama  65  Dec 29, 1995    Richard III   
2872              Documentary  12  Oct 24, 2008         Heroes   
2873          Biography,Drama  33  Mar 29, 2019      Unplanned   
2874                 Thriller  48  Jan 14, 2000  The Terrorist   

     production_budget domestic_gross worldwide_gross  averagerating  numvotes  
0          $45,000,000             $0         $73,706            1.9      8248  
1             $200,000     $1,109,808      $1,165,996            7.5        24  
2             $200,000     $1,109,808      $1,165,996            6.1     14828  
3          $25,000,000       $720,828      $9,313,302            6.1     37886  
4          $25,000,000       $720,828      $9,313,302            6.0         6  
...                ...            ...             ...            ...       ...  
2870          $500,000       $307,631        $308,793            6.6        27  
2871        $9,200,000     $2,684,904      $4,199,334            9.1        28  
2872          $400,000       $655,538        $655,538            7.3         7  
2873        $6,000,000    $18,107,621     $18,107,621            6.3      5945  
2874           $25,000       $195,043        $195,043            6.0         6  

[2875 rows x 13 columns]

In [32]:
# Dropping the rows with null values
final_merge_df.dropna( axis = 0 , inplace = True)

In [33]:
final_merge_df

movie_id  primary_title   original_title  start_year  \
0     tt0249516     Foodfight!       Foodfight!        2012   
2     tt3844362  The Overnight    The Overnight        2015   
3     tt0337692    On the Road      On the Road        2012   
4     tt4339118    On the Road      On the Road        2014   
5     tt5647250    On the Road      On the Road        2016   
...         ...            ...              ...         ...   
2870  tt8976472    Never Again  Ei koskaan enää        2018   
2871  tt8680254    Richard III      Richard III        2016   
2872  tt8824064         Heroes           Heroes        2019   
2873  tt9024106      Unplanned        Unplanned        2019   
2874  tt9248762  The Terrorist    The Terrorist        2018   

                       genres  id  release_date          movie  \
0     Action,Animation,Comedy  26  Dec 31, 2012     Foodfight!   
2              Comedy,Mystery  21  Jun 19, 2015  The Overnight   
3     Adventure,Drama,Romance  17  Mar 22, 2013    On the Road   
4                       Drama  17  Mar 22, 2013    On the Road   
5                       Drama  17  Mar 22, 2013    On the Road   
...                       ...  ..           ...            ...   
2870        Documentary,Music  47  Jul 12, 2002    Never Again   
2871                    Drama  65  Dec 29, 1995    Richard III   
2872              Documentary  12  Oct 24, 2008         Heroes   
2873          Biography,Drama  33  Mar 29, 2019      Unplanned   
2874                 Thriller  48  Jan 14, 2000  The Terrorist   

     production_budget domestic_gross worldwide_gross  averagerating  numvotes  
0          $45,000,000             $0         $73,706            1.9      8248  
2             $200,000     $1,109,808      $1,165,996            6.1     14828  
3          $25,000,000       $720,828      $9,313,302            6.1     37886  
4          $25,000,000       $720,828      $9,313,302            6.0         6  
5          $25,000,000       $720,828      $9,313,302            5.7       127  
...                ...            ...             ...            ...       ...  
2870          $500,000       $307,631        $308,793            6.6        27  
2871        $9,200,000     $2,684,904      $4,199,334            9.1        28  
2872          $400,000       $655,538        $655,538            7.3         7  
2873        $6,000,000    $18,107,621     $18,107,621            6.3      5945  
2874           $25,000       $195,043        $195,043            6.0         6  

[2867 rows x 13 columns]

In [34]:
# Checking for null values
final_merge_df.isna().sum()

movie_id             0
primary_title        0
original_title       0
start_year           0
genres               0
id                   0
release_date         0
movie                0
production_budget    0
domestic_gross       0
worldwide_gross      0
averagerating        0
numvotes             0
dtype: int64

There are no more null values in the final_merge_df dataset

### 2: extraneous values 

In [39]:
for col in final_merge_df.columns:
    print(col,"\n", final_merge_df[col].value_counts(normalize = True),"\n\n")

movie_id 
 tt2093100    0.001046
tt4047846    0.001046
tt2545384    0.001046
tt2224026    0.001046
tt2075392    0.001046
               ...   
tt0938283    0.000349
tt1832471    0.000349
tt3533916    0.000349
tt1441326    0.000349
tt2234025    0.000349
Name: movie_id, Length: 2745, dtype: float64 


primary_title 
 Home                                           0.008371
Brothers                                       0.005581
The Gift                                       0.005581
Crossroads                                     0.004186
Hamlet                                         0.004186
                                                 ...   
Step Up 3D                                     0.000349
Eli                                            0.000349
Freeheld                                       0.000349
War Dogs                                       0.000349
Pirates of the Caribbean: On Stranger Tides    0.000349
Name: primary_title, Length: 2126, dtype: float64 


original_title

There are no more missing or unknown values in our dataset

### 3:Duplicate values

In [52]:
final_merge_df.duplicated().sum()

0

The dataset has no missing values

### 3. Outliers

In [53]:
final_merge_df.describe()

start_year           id  averagerating      numvotes
count  2867.000000  2867.000000    2867.000000  2.867000e+03
mean   2013.916638    50.967213       6.249111  6.646528e+04
std       2.547187    28.680842       1.185953  1.344493e+05
min    2010.000000     1.000000       1.600000  5.000000e+00
25%    2012.000000    27.000000       5.600000  1.465000e+02
50%    2014.000000    51.000000       6.400000  7.999000e+03
75%    2016.000000    76.000000       7.100000  7.576750e+04
max    2019.000000   100.000000       9.300000  1.841066e+06

In [54]:
final_merge_df["production_budget"] = final_merge_df["production_budget"].replace({"\$":"",",":""},regex = True).astype(float)
final_merge_df["domestic_gross"] = final_merge_df["domestic_gross"].replace({"\$":"",",":""},regex = True).astype(float)
final_merge_df["worldwide_gross"] = final_merge_df["worldwide_gross"].replace({"\$":"",",":""},regex = True).astype(float)


In [55]:
final_merge_df.describe()

start_year           id  production_budget  domestic_gross  \
count  2867.000000  2867.000000       2.867000e+03    2.867000e+03   
mean   2013.916638    50.967213       3.428709e+07    4.301514e+07   
std       2.547187    28.680842       4.767275e+07    7.460932e+07   
min    2010.000000     1.000000       1.400000e+03    0.000000e+00   
25%    2012.000000    27.000000       5.000000e+06    4.963625e+05   
50%    2014.000000    51.000000       1.700000e+07    1.637607e+07   
75%    2016.000000    76.000000       4.000000e+07    5.178837e+07   
max    2019.000000   100.000000       4.250000e+08    7.605076e+08   

       worldwide_gross  averagerating      numvotes  
count     2.867000e+03    2867.000000  2.867000e+03  
mean      1.028804e+08       6.249111  6.646528e+04  
std       2.039293e+08       1.185953  1.344493e+05  
min       0.000000e+00       1.600000  5.000000e+00  
25%       2.259093e+06       5.600000  1.465000e+02  
50%       2.996005e+07       6.400000  7.999000e+03  
75%       1.012707e+08       7.100000  7.576750e+04  
max       2.776345e+09       9.300000  1.841066e+06

Since domestic_gross , worldwide_gross and production_budget vary, we cannot check for outliers, however,the  average rating has no outliers

#  DATA ANALYSIS